sotodlib/tod_ops/azss.py

```diff
@@ -235,43 +270,134 @@ def get_azss(aman, signal='signal', az=None, range=None, bins=100, flags=None,
+                                        apodize_edges_samps, apodize_flags, apodize_flags_samps,
+                                        method=method, max_mode=max_mode)
+        azss_right.add_axis(aman.samps)
+        azss_right.wrap('mask', right_mask, [(0, 'samps')])
+        azss_stats = core.AxisManager(aman.dets)
+        azss_stats.wrap('azss_stats_left', azss_left)
+        azss_stats.wrap('azss_stats_right', azss_right)
+        azss_stats.wrap('left_right', left_right)
+    else:
+        azss_stats = _prepare_azss_stats(aman, signal, az, frange, bins, flags, apodize_edges,
+                                        apodize_edges_samps, apodize_flags, apodize_flags_samps,
+                                        method=method, max_mode=max_mode)
+        azss_stats.wrap('left_right', left_right)
+
+    model_left, model_right, model = None, None, None
+    if merge_model or subtract_in_place:
+        if left_right:
+            azss_stats, model_left, model_right = get_model_sig_tod(aman, azss_stats, az)
+            if merge_model:
+                aman.wrap(azss_model_name+'_left', model_left, [(0, 'dets'), (1, 'samps')])
+                aman.wrap(azss_model_name+'_right', model_right, [(0, 'dets'), (1, 'samps')])
+            if subtract_in_place:
+                if signal_name is None:
+                    lmask = left_mask.mask()
+                    signal[:,lmask] -= model_left[:,lmask].astype(signal.dtype)
+                    rmask = right_mask.mask()
+                    signal[:,rmask] -= model_right[:,rmask].astype(signal.dtype)
+                else:
+                    lmask = left_mask.mask()
+                    aman[signal_name][:,lmask] -= model_left[:,lmask].astype(aman[signal_name].dtype)
+                    rmask = right_mask.mask()
+                    aman[signal_name][:,rmask] -= model_right[:,rmask].astype(aman[signal_name].dtype)
+        else:
+            azss_stats, model, _ = get_model_sig_tod(aman, azss_stats, az)
+            if merge_model:
+                aman.wrap(azss_model_name, model, [(0, 'dets'), (1, 'samps')])
+            if subtract_in_place:
+                if signal_name is None:
+                    signal -= model.astype(signal.dtype)
+                else:
+                    aman[signal_name] -= model.astype(aman[signal_name].dtype)
+
     if merge_stats:
         aman.wrap(azss_stats_name, azss_stats)
-    if merge_model:
-        aman.wrap(azss_model_name, model_sig_tod, [(0, 'dets'), (1, 'samps')])
-    if subtract_in_place:
-        aman[signal_name] = np.subtract(signal, model_sig_tod, dtype='float32')
-    return azss_stats, model_sig_tod
 
-def subtract_azss(aman, signal='signal', azss_template_name='azss_model',
-                  subtract_name='azss_remove', in_place=False, remove_template=True):
+    if left_right:
+        return azss_stats, model_left, model_right
+    else:
+        return azss_stats, model
+
+def get_model_sig_tod(aman, azss_stats, az=None):
```

Comment:
I was thinking that these could be used outside of azss.py since azss_stats are stored in a sparser format (just shape n_dets x n_az_bins at most) if one wants to regenerate the full n_dets x n_samps model for subtraction or plotting later they'll want to use these functions.